src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -0,0 +1,386 @@
+                hasError = true;
+            }
+
+            BinaryOperatorKind elementOperatorKind = hasError ? kind : kind.WithType(BinaryOperatorKind.Dynamic);
+            TypeSymbol dynamicType = Compilation.DynamicType;
+
+            // We'll end up dynamically invoking operators true and false, so no result conversion. We'll deal with that during lowering.
+            return new TupleBinaryOperatorInfo.Single(dynamicType, dynamicType, elementOperatorKind,
+                methodSymbolOpt: null, boolConversion: Conversion.Identity, boolOperator: default);
+        }
+
+        private TupleBinaryOperatorInfo BindTupleBinaryOperatorNestedInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            ImmutableArray<BoundExpression> left, ImmutableArray<BoundExpression> right,
+            bool nullable, DiagnosticBag diagnostics)
+        {
+            int length = left.Length;
+            Debug.Assert(length == right.Length);
+
+            var operatorsBuilder = ArrayBuilder<TupleBinaryOperatorInfo>.GetInstance(length);
+
+            for (int i = 0; i < length; i++)
+            {
+                operatorsBuilder.Add(BindTupleBinaryOperatorInfo(node, kind, left[i], right[i], diagnostics));
+            }
+
+            var compilation = this.Compilation;
+            TypeSymbol leftTupleType = MakeConvertedType(operatorsBuilder, node.Left, left, nullable, compilation, diagnostics, isRight: false);
+            TypeSymbol rightTupleType = MakeConvertedType(operatorsBuilder, node.Right, right, nullable, compilation, diagnostics, isRight: true);
+
+            return new TupleBinaryOperatorInfo.Nested(operatorsBuilder.ToImmutableAndFree(), leftTupleType, rightTupleType);
+        }
+
+        private static int? GetTupleCardinality(BoundExpression expr)
+        {
+            if (expr.Kind == BoundKind.TupleLiteral)
+            {
+                var tuple = (BoundTupleLiteral)expr;
+                return tuple.Arguments.Length;
+            }
+
+            TypeSymbol type = expr.Type;
+            if (type is null)
+            {
+                return null;
+            }
+
+            type = type.StrippedType();
+
+            if (type.IsTupleType)
+            {
+                return type.TupleElementTypes.Length;
+            }
+
+            return null;
+        }
+
+        private static ImmutableArray<BoundExpression> GetTupleArgumentsOrPlaceholders(BoundExpression expr)
+        {
+            if (expr.Kind == BoundKind.TupleLiteral)
+            {
+                return ((BoundTupleLiteral)expr).Arguments;
+            }
+
+            // placeholder bound nodes with the proper types are sufficient to bind the element-wise binary operators
+            return expr.Type.StrippedType().TupleElementTypes
+                .SelectAsArray(t => (BoundExpression)new BoundTupleOperandPlaceholder(expr.Syntax, t));
```

Comment:
Pass `expr.Syntax` as argument to avoid closure class: `.SelectAsArray((t, s) => (BoundExpression)new BoundTupleOperandPlaceholder(s, t), expr.Syntax)`.